### DISCUSSION TITLE
[INFO GRT-0101] Running extra iterations to remove overflow - taking a long time

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2889] [General]
What does this step actually do?

Any hints on how to figure out what is going on here, what to look at?

When I zoom in on the design, I see an ocean of thinly spread fiddlybits:

![image](https://user-images.githubusercontent.com/2798822/218471305-7312e5f8-afbf-454d-9dbd-87a75d5197e5.png)

**Comment by rovinski** [comment]
GRT is the global router. It means your design is very dense (at least within some areas). It is attempting to assign nets to routing tracks, but there are too many nets within an area (overflow) so it is trying to reroute nets to avoid this.

GRT runs for a fixed number of iterations before giving up and erroring if it can't resolve the overflow.

The best first thing to check is the global routing heatmap to see congestion hotspots. In your case, it is likely that the density is simply too high and you need to reduce the placement density %.

**Reply by oharboe** [reply]
```
[INFO GRT-0199] Via filling finished.
[INFO GRT-0111] Final number of vias: 4339652
[INFO GRT-0112] Final usage 3D: 25787116
[ERROR GRT-0119] Routing congestion too high. Check the congestion heatmap in the GUI and load congestion.rpt in the DRC viewer.
Error: global_route.tcl, 18 GRT-0119
Elapsed time: 2:02:16[h:]min:sec. CPU time: user 7334.77 sys 1.21 (99%). Peak memory: 2805080KB.
```

Quite :-)

![image](https://user-images.githubusercontent.com/2798822/218686387-9cf7d4d7-d458-404b-9466-749881bb4f7b.png)

100% congestion everywhere in heatmap:

![image](https://user-images.githubusercontent.com/2798822/218686699-e0915ec8-e6dd-4200-956f-68958796456e.png)

**Reply by rovinski** [reply]
I agree with @maliberty that it's likely there is some combination of 1) design placement is too dense and 2) the timing is overconstrained so too many buffers are being inserted.

**Reply by oharboe** [reply]
Yes: tweaking those two things, it gets to detailed routing, but after 24 hours it hasn't completed. It's still chugging along.

The Quartus FPGA tools gets through this in 10 minutes or so. I believe I need to introduce some macros here so that it scales better.

```
[INFO DRT-0195] Start 3rd optimization iteration.
    Completing 10% with 186730 violations.
    elapsed time = 01:00:31, memory = 56934.16 (MB).
    Completing 20% with 163222 violations.
    elapsed time = 02:56:06, memory = 56952.45 (MB).
    Completing 30% with 151449 violations.
    elapsed time = 03:55:55, memory = 59168.17 (MB).
    Completing 40% with 134227 violations.
    elapsed time = 06:09:38, memory = 57391.38 (MB).
    Completing 50% with 118763 violations.
    elapsed time = 08:13:37, memory = 54116.43 (MB).
    Completing 60% with 103749 violations.
    elapsed time = 09:36:43, memory = 58544.92 (MB).
```

**Reply by Emrys025** [reply]
Were you able to solve the problem?

**Comment by maliberty** [comment]
You should also check how many buffers are being inserted as it possible that can cause extra congestion if you have very bad timing.
